src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/FieldBuilderImpl.cs

```diff
@@ -74,8 +74,12 @@ internal static void ValidateDefaultValueType(object? defaultValue, Type destina
                     {
                         underlyingType = enumBldr.GetEnumUnderlyingType();
 
-                        if (sourceType != enumBldr._typeBuilder.UnderlyingSystemType && sourceType != underlyingType)
+                        if (sourceType != enumBldr._typeBuilder.UnderlyingSystemType &&
+                            sourceType != underlyingType &&
+                            sourceType.IsEnum && sourceType.GetEnumUnderlyingType() != underlyingType)
```

Comment:
```suggestion
                        if (sourceType != enumBldr._typeBuilder.UnderlyingSystemType &&
                            sourceType != underlyingType &&
                            sourceType.IsEnum &&
                            sourceType.GetEnumUnderlyingType() != underlyingType)
```

Can we also get a comment on this predicate? It isn't intuitive.

Reply:
> Can we also get a comment on this predicate? It isn't intuitive.

Added `If the source type is an enum, should not throw when the underlying types match`